### PR TITLE
 depends: set several missing build tags 

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by vtnerd** [comment]
One giant benefit of continually building is getting quicker feedback if a platform is failing. It could be a rush during release time if a replacement process isn't in place.

Boost (used to?) has a test matrix where volunteers from the community setup a unique environment to compile + run tests and report back to a web interface. The testing is only done against commits and not against each PR - which reduces load considerably. Our commits are "batched" too, so they only need to run once a week (or less) after a group of commits. I should be able to obtain the code (probably Python) for both the server and the builds that ran this matrix. It generated static html files IIRC. 

If Gitian is doing cross-compiles, then these systems don't have to be lock-down secure for release building, they only need to sanity checking periodically (with a rush during the lead-up to release).

**Comment by vtnerd** [comment]
Whooops wrong tab entirely. Deleting above.

File: contrib/depends/toolchain.cmake.in [hunk 1]
```diff
@@ -136,11 +136,21 @@ endif()
...
 endif()
 
 if(ARCHITECTURE STREQUAL "x86_64")
+  if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    set(BUILD_TAG "linux-x64")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
+    set(BUILD_TAG "freebsd-x64")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    set(BUILD_TAG "win-x64")
+  endif()
```
Comment (hyc) [inline contrib/depends/toolchain.cmake.in]: Is a case for MacOSX needed here too?

Comment (moneromooo-monero) [reply contrib/depends/toolchain.cmake.in]: The mac one is set in the existing Darwin clause already.

**Pull request merged**